src/main/java/org/jboss/tools/intellij/openshift/actions/component/DebugComponentAction.java

```diff
@@ -0,0 +1,155 @@
+        ApplicationNode applicationNode = (ApplicationNode) componentNode.getParent();
+        try {
+            Odo odo = ((ApplicationsRootNode) componentNode.getRoot()).getOdo();
+            LazyMutableTreeNode projectNode = (LazyMutableTreeNode) applicationNode.getParent();
+            ApplicationsRootNode root = (ApplicationsRootNode) applicationNode.getRoot();
+            ComponentInfo info = odo.getComponentInfo(root.getClient(), projectNode.toString(), applicationNode.toString(), component.getName());
+            return isDebuggable(info.getComponentTypeName());
+        } catch (IOException e) {
+            UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Debug"));
+        }
+        return false;
+    }
+
+    private boolean isPushed(Component component) {
+        return component.getState() == ComponentState.PUSHED;
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Odo odo) {
+        ComponentNode componentNode = (ComponentNode) selected;
+        Component component = (Component) componentNode.getUserObject();
+        ApplicationNode applicationNode = (ApplicationNode) componentNode.getParent();
+        LazyMutableTreeNode projectNode = (LazyMutableTreeNode) applicationNode.getParent();
+
+        Project project = anActionEvent.getData(CommonDataKeys.PROJECT);
+
+        RunManager runManager = RunManager.getInstance(project);
+        ConfigurationType configurationType = getConfigurationType();
+        String configurationName = component.getName() + " Remote Debug";
+        //lookup if existing config already exist, based on name and type
+        runSettings = runManager.findConfigurationByTypeAndName(configurationType.getId(), configurationName);
+        if (runSettings == null) {
+            runSettings = runManager.createConfiguration(
+                    configurationName, configurationType.getConfigurationFactories()[0]);
+            ServerSocket serverSocket = null;
+            try {
+                serverSocket = new ServerSocket(0); // find an available port and use it
+                port = serverSocket.getLocalPort();
+                serverSocket.close();
+                initConfiguration(runSettings.getConfiguration(), port);
+            } catch (IOException e) {
+                UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Odo Debug"));
+                return;
+            }
+
+        } else {
+            port = getPortFromConfiguration(runSettings.getConfiguration());
+            if (port == -1) {
+                UIHelper.executeInUI(() -> Messages.showErrorDialog("Error when retrieving local port from configuration.", "Odo Debug"));
+                return;
+            }
+
+        }
+        ExecHelper.submit(() -> {
+            try {
+                // test if the port is still available before running odo
```

Comment:
Without the odo debug status, it seems complex to implement it like it now. 
this test is only to verify if odo and the debugger can connect together. 
If the user terminate the debugger without closing the terminal session where odo is running, there will be an error saying the port is already used, but there is no more indication of what process is using that port. 
I agree that if odo debug status return that odo is running on that port, i should only run the debugger. ( and also test if the debugger is not already running )

Reply:
I don't understand why the proposed algorithm does not emulate the odo status

Reply:
it can. i'll change the algorithm.
My concern was about if the port is used by another process while odo is not running. But this case should not apply often.